FIX #28347 FIX #28962 FIX #29224  FIX #29035

sonikf opened the pull request:
This feature is currently quite broken, first try to remedy.
Also in #29035 payment terms and other client attributes are carried over.

sonikf committed:
FIX #28347 FIX #28962 FIX #29224

sonikf committed:
fix empty line

eldy commented on htdocs/fourn/commande/card.php:
```diff
@@ -1361,7 +1385,7 @@
 									$product_fourn_price_id,
 									$ref_supplier,
-									$lines[$i]->remise_percent,
+									0,     // $lines[$i]->remise_percent must not be carried over from client to vendor
```
The supplier order can be created from a supplier proposal. In such case, we must keep all information of the proposal. So we must use here 0 only to create supplier order from a customer order.

sonikf committed:
Update card.php

sonikf committed:
FIX #29035

sonikf committed:
Merge branch '19.0' into sonikf-patch-supplier-order-from-client-order

eldy commented on htdocs/fourn/commande/card.php:
```diff
@@ -1333,10 +1334,35 @@
+											$sql = 'SELECT price, unitprice, tva_tx, remise_percent, ref_fourn';
+											$sql .= ' FROM '.MAIN_DB_PREFIX.'product_fournisseur_price';
+											$sql .= ' WHERE fk_product = '.((int) $li->fk_product);
+											$sql .= ' AND fk_soc = '.((int) $object->socid);
```
Can you check if adding here
`$sql .= " AND entity IN ('".getEntity("product_fournisseur_price").")";`
Your fixis still ok (i think we need this for multicompany support)

sonikf replied:
It works!

sonikf committed:
add multicompany support in creation of supplier order from client order

sonikf committed:
Update card.php

Assign a subprice value of 0 to user selected lines without a vendor product price when creating a supplier order from a client order
